### PR TITLE
Boris/memegen adapter1.0

### PR DESCRIPTION
**Pull request by borisjaskerovich**
at this version i am not using prebid. globals.
i added some new parameters to the bid response as required by the api.
most important: requestId->connect to bidId of the request.

### CODE REVIEW

File: modules/exampleBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
..._EVLVXwpqU3A_GQAAAKBH4eo_IZrniHyXUnc_Kd5xio7k8n8_MQAAAEDhepQ_MKHJ-wU40RhAlgNIAlCUoa84WOzWV2AAaIqycXj7kwWAAQGKAQNVU0SSAQNVU0SYAawCoAHYBKgBAbABALgBAcABBcgBAtABANgBAOABAPABAIoCWnVmKCdhJywgMjcyOTQ4NiwgMTU0NDQzNTk4Myk7dWYoJ3InLCAxMTgyMTQ4MDQsIDE1NDQ0MzU5ODMpO3VmKCdjJywgMjU2ODYzOTUsIDE1NDQ0MzU5ODMpO5IC8QEhb2pObHdnajc0cDhNRUpTaHJ6Z1lBQ0RzMWxjd0FEZ0FRQUJJbGdOUW9jbjdCVmdBWUhob0FIQUFlQUNBQVNLSUFTaVFBUUdZQVFHZ0FRR29BUU93QVFDNUFhVEhMNVhsOG44X3dRR2t4eS1WNWZKX1A4a0JBQUFBQUFBQThEX1pBUUFBQUFBQUFQQV80QUVBOVFFQUFBQUFtQUlBb0FJQXRRSUEBLQh2UUkBB_BMQXdBSUF5QUlBNEFJQTZBSUEtQUlBZ0FNQmtBTUFtQU1CcUFQNzRwOE11Z01KUmxKQk1UbzBNVFEyNEFQTkFnLi6aAmEhbEJIWk5BajcBKABFDfRcN05aWElBQW9BREg4cWZIU1RXSlFQem9KLkQAEFFNMENTHbAAVREMDEFBQVcdDPQYAdgC50PgAp2PRuoCOWh0dHA6Ly9tZWRpYW1hcnQudHYvc2FzL3Rlc3RzL0Rlc2t0b3AvY2Flc2FyL2RmcHRlc3QuaHRtbIADAIgDAZADAJgDF6ADAaoDAMADrALIAwDYAwDgAwDoAwD4AwGABACSBA0vdXQvdjMvcHJlYmlkmAQAogQNMzEuMTU0LjE2OC4zNKgE_-ELsgQQCAAQARisAiDYBCgAMAA4ArgEAMAEAMgEANIEDTQwNiNGUkExOjQxNDbaBAIIAeAEAfAElKGvOIgFAZgFAKAF____________AcAFAMkFAAAAAAAA8D_SBQkJAAAAAAAAAADYBQHgBQHwBdpK-gUECAAQAJAGAJgGALgGAMEGAAAABTAcyAYA2gYWChAFMTgAAAAAAAAAAAAAABAAGAA.&s=f8188a3c4f66ff8dacda42b34fa9793a28b168a6\"></div>",mediaType:"banner"}];
```
Comment (shlomisut) [inline modules/exampleBidAdapter.js:43]: what's this creative you're loading ?

File: modules/exampleBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+   *
+   * @param {SyncOptions} syncOptions Which user syncs are allowed?
+   * @param {ServerResponse[]} serverResponses List of server's responses.
+   * @return {UserSync[]} The user syncs which should be dropped.
+   */
+  getUserSyncs: function(syncOptions, serverResponses) {
+    if (syncOptions.iframeEnabled) {
+      return [{
+        type: 'iframe',
+        url: '//acdn.adnxs.com/ib/static/usersync/v3/async_usersync.html'
```
Comment (shlomisut) [inline modules/exampleBidAdapter.js]: why do you have this set ?

File: modules/memegenBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+  let params=bidRequest.url.split('?')[1];
+  var obj = {};
+  var pairs = params.split('&');
+  try{
+    for(var i in pairs){
+      var split = pairs[i].split('=');
+      obj[decodeURIComponent(split[0])] = decodeURIComponent(split[1]);
+    }
+  }catch (e) {
+    console.log(e);
```
Comment (shlomisut) [inline modules/memegenBidAdapter.js]: remove

File: modules/exampleBidAdapter.js (hunk 1 above)
Comment (borisjaskerovich) [reply modules/exampleBidAdapter.js:43]: appnexus

File: modules/exampleBidAdapter.js (hunk 2 above)
Comment (borisjaskerovich) [reply modules/exampleBidAdapter.js]: i will remove it

**Comment by ghost** [comment]
Build Passed

**Comment by borisjaskerovich** [comment]
I copied most of the code from the old version. maybe i should not use vars at this version.
I think prebid new version probably handles it.

**Comment by ghost** [comment]
Build Passed

**Comment by ghost** [comment]
Build Passed

**Comment by ghost** [comment]
Build Passed

**Review by shlomisut — APPROVED** [review]

**Pull request merged**